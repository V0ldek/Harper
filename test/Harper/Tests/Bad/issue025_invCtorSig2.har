ref RefBox a = {
    data = {
        a :: a;
    }

    ctor :: a -> RefBox a;
    ctor a = {
        this.data.a := a;
        return ();
    };

    get :: impure -> a;
    get = this.data.a;

    set :: a -> impure -> ();
    set a = {
        this.data.a := a;
    };
};

main :: Integer;
main = {
    (b :: RefBox Integer) = RefBox.ctor 42;
    return b.get ();
};