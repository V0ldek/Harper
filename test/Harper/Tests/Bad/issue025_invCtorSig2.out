Error: constructor of `RefBox` has an invalid signature `a -> RefBox a`. Constructors must be impure functions returning an instance of the type.
During evaluation of:
  ctor :: a -> RefBox a
Located at line 6 column 4

Execution terminated with an error: type error.